
Disable some assembler optimizations due to gcc "bugs", as suggested here:

  http://lists.gnu.org/archive/html/qemu-devel/2004-07/msg00436.html

  - Rene Rebe <rene@exactcode.de>

--- qemu-0.6.0/target-i386/op.c	2004-07-10 20:20:09.000000000 +0200
+++ qemu-0.6.0/target-i386/op.c	2004-11-10 12:16:35.080626836 +0100
@@ -22,7 +22,7 @@
    register constraints so they cannot be used in any C code. gcc 3.3
    does not seem to be able to handle some constraints in rol
    operations, so we disable it. */
-#if !(__GNUC__ == 3 && __GNUC_MINOR__ == 3)
+#if !(__GNUC__ == 3 && __GNUC_MINOR__ == 2) && !(__GNUC__ == 3 && __GNUC_MINOR__ == 3) && !(__GNUC__ == 3 && __GNUC_MINOR__ == 4)
 #define ASM_SOFTMMU
 #endif
 #include "exec.h"
